### PR TITLE
Remove git_release functionality

### PR DESCRIPTION
**Pull request by dlpzx**
### Feature or Bugfix
- Bugfix

### Detail
Deprecate `git_release` functionality. It is a feature that results in errors and unnecessary CICD pipeline executions. We could fix it, but we are not aware of any customer using it. In case needed it could be re-added.

Depending on the outcome of this PR I will open a PR to update the docs

### Relates
- #617 

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by petrkalos — APPROVED** [review]

File: deploy/stacks/pipeline.py [hunk 1]
```diff
@@ -1013,42 +975,3 @@ def user_guide_pre_build_alb(self, repository_name):
...
-                                    'set -eu',
-                                    f'aws codeartifact login --tool pip --repository {self.codeartifact.codeartifact_pip_repo_name} --domain {self.codeartifact.codeartifact_domain_name} --domain-owner {self.codeartifact.domain.attr_owner}',
-                                    'python -m venv env',
-                                    '. env/bin/activate',
-                                    'pip install git-remote-codecommit',
-                                    'mkdir release && cd release',
-                                    f'git clone codecommit::{self.region}://dataall',
-                                    'cd dataall',
-                                    f'git checkout {self.git_branch}',
-                                    f'make version-minor branch={self.git_branch}',
```
Comment (petrkalos) [inline deploy/stacks/pipeline.py:1043]: should you also remove the `Makefile` target?

File: deploy/stacks/pipeline.py [hunk 2]
```diff
@@ -206,9 +206,6 @@ def __init__(
             else:
                 self.set_albfront_stage(target_env, repository_name)
 
-        if self.node.try_get_context('git_release'):
-            self.set_release_stage()
```
Comment (noah-paige) [inline deploy/stacks/pipeline.py:210]: Can we also delete `self.git_project_role()` and associated policies?

Comment (dlpzx) [reply deploy/stacks/pipeline.py:210]: That is already done right?

File: deploy/stacks/pipeline.py (hunk 1 above)
Comment (dlpzx) [reply deploy/stacks/pipeline.py:1043]: Good idea. Done!

**Review by noah-paige — APPROVED** [review]
lgtm

**Pull request merged**